src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/AbstractMonsterSpawner.java

```diff
@@ -83,12 +83,11 @@ public ItemStack getItemForEntityType(@Nonnull EntityType type) {
         ItemMeta meta = item.getItemMeta();
 
         // Fixes #2583 - Proper NBT handling of Spawners
-        if (meta instanceof BlockStateMeta) {
-            BlockStateMeta stateMeta = (BlockStateMeta) meta;
+        if (meta instanceof BlockStateMeta stateMeta) {
             BlockState state = stateMeta.getBlockState();
 
-            if (state instanceof CreatureSpawner) {
-                ((CreatureSpawner) state).setSpawnedType(type);
+            if (state instanceof CreatureSpawner creatureSpawner) {
+                creatureSpawner.setSpawnedType(type);
```

Comment:
We do not always have to be so literal about it.
```suggestion
            if (state instanceof CreatureSpawner spawner) {
                spawner.setSpawnedType(type);
```